Interesting, it seems `100` is a value too low for the CI. All the tests are passing locally, but on the CI some of the browser tests are failing:

```
⎯⎯⎯⎯⎯⎯⎯ Failed Tests 2 ⎯⎯⎯⎯⎯⎯⎯

 FAIL   node  src/stack.test.ts > tcp > tcp packets are sent immediately without delay
AssertionError: expected 100.12071800000012 to be less than 100
 ❯ src/stack.test.ts:882:21
    880|     // Without tcp_output, there would be a significant delay (~300ms)
    881|     const elapsed = endTime - startTime;
    882|     expect(elapsed).toBeLessThan(100);
       |                     ^
    883|   });
    884| });

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/2]⎯

 FAIL   browser (webkit)  src/stack.test.ts > tcp > tcp packets are sent immediately without delay
AssertionError: expected 100 to be less than 100
 ❯ src/stack.test.ts:882:33
    880|     // Without tcp_output, there would be a significant delay (~300ms)
    881|     const elapsed = endTime - startTime;
    882|     expect(elapsed).toBeLessThan(100);
       |                                 ^
    883|   });
    884| });
```

Let me increase the threshold for testing on the CI.

--- a/packages/tcpip/src/stack.test.ts
+++ b/packages/tcpip/src/stack.test.ts
@@ -879,7 +879,7 @@ describe('tcp', () => {
     // Check timing - with tcp_output enabled, this should be very fast
     // Without tcp_output, there would be a significant delay (~300ms)
     const elapsed = endTime - startTime;
-    expect(elapsed).toBeLessThan(100);
+    expect(elapsed).toBeLessThan(200);
   });
 });
 

--- a/packages/tcpip/src/stack.test.ts
+++ b/packages/tcpip/src/stack.test.ts
@@ -844,16 +844,28 @@ describe('tcp', () => {
   });
 
   test('tcp packets are sent immediately without delay', async () => {
-    const stack = await createStack();
+    const stack1 = await createStack();
+    const stack2 = await createStack();
 
-    const listener = await stack.listenTcp({
-      host: '127.0.0.1',
+    const tun1 = await stack1.createTunInterface({
+      ip: '192.168.1.1/24',
+    });
+
+    const tun2 = await stack2.createTunInterface({
+      ip: '192.168.1.2/24',
+    });
+
+    // Connect the two interfaces
+    tun1.readable.pipeTo(tun2.writable);
+    tun2.readable.pipeTo(tun1.writable);
+
+    const listener = await stack2.listenTcp({
       port: 8080,
     });
 
     const [outbound, inbound] = await Promise.all([
-      stack.connectTcp({
-        host: '127.0.0.1',
+      stack1.connectTcp({
+        host: '192.168.1.2',
         port: 8080,
       }),
       nextValue(listener),
@@ -866,20 +878,20 @@ describe('tcp', () => {
 
     // Record start time
     const startTime = performance.now();
-    
+
     // Write and read immediately
     await outboundWriter.write(data);
     const received = await inboundReader.read();
 
     // Record end time
     const endTime = performance.now();
-    
+
     expect(received.value).toStrictEqual(data);
-    
+
     // Check timing - with tcp_output enabled, this should be very fast
     // Without tcp_output, there would be a significant delay (~300ms)
     const elapsed = endTime - startTime;
-    expect(elapsed).toBeLessThan(200);
+    expect(elapsed).toBeCloseTo(0, -1);
   });
 });
 

